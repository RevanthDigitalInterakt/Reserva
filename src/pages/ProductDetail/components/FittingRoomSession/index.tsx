import React from "react";
import { View } from "react-native";

import SizeGuide from "./components/SizeGuide";
import FooterDoris from "./components/FooterDoris";
import { Divider } from "../../../../components/Divider/Divider";
import styles from "./styles";
import ButtonDoris from "./components/ButtonDoris";
import testProps from "../../../../utils/testProps";

interface IFittingRoomSession {
  categoryTree?: { name: string }[] | null;
  productId?: string;
  productEan?: string;
  isValidProductDoris: boolean;
}

export default function FittingRoomSession({
  categoryTree,
  productId,
  productEan,
  isValidProductDoris,
}: IFittingRoomSession) {
  return (
    <View {...testProps("fitting_room_session")}>
      {isValidProductDoris && !!categoryTree?.length && (
        <Divider variant="fullWidth" my="xs" />
      )}
      <View style={styles.childContainer}>
        {isValidProductDoris && (
          <ButtonDoris
            enabledBtnFullDoris={!categoryTree?.length}
            productEan={productEan}
            productId={productId}
          />
        )}

        {!!categoryTree?.length && (
          <SizeGuide
            categoryTree={categoryTree}
            productId={productId}
            enabledBtnFullSizeGuide={!isValidProductDoris}
          />
        )}
      </View>

      {isValidProductDoris && (
        <FooterDoris enabledBtnFullDoris={!!categoryTree?.length} />
      )}

      <Divider variant="fullWidth" my="xs" />
    </View>
  );
}
