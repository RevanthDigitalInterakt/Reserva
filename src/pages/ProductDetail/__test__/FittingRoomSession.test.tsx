import React from 'react';
import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react-native';
import FittingRoomSession from '../components/FittingRoomSession';
import ButtonDoris from '../components/FittingRoomSession/components/ButtonDoris';

jest.mock('../../../zustand/useBagStore/useBagStore', () => ({
  useBagStore: () => ({
    orderformId: '123',
  }),
}));

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockedNavigate }),
}));

const categoryTreeValid = [{ name: 'Reserva' }, { name: 'Masculino' }, { name: 'polos' }];

const productIdValid = '45660';

const eanValid = '0082519026';

const Component = (
  <FittingRoomSession
    categoryTree={categoryTreeValid}
    productEan={eanValid}
    productId={productIdValid}
    isValidProductDoris
  />
);

describe('FittingRoomSession', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should snapshot', () => {
    render(Component);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should initial render component', () => {
    const { getByTestId } = render(Component);
    const component = getByTestId('com.usereserva:id/fitting_room_session');
    const componentBtnDoris = getByTestId('com.usereserva:id/component_button_doris');
    const btnDoris = getByTestId('com.usereserva:id/button_doris');
    const txtNew = getByTestId('com.usereserva:id/txt_new');
    const buttonSizeGuide = getByTestId('com.usereserva:id/size_guides_button_product_details');
    const footerDoris = getByTestId('com.usereserva:id/footer_doris');

    expect(component).toBeOnTheScreen();
    expect(componentBtnDoris).toBeOnTheScreen();
    expect(btnDoris).toBeOnTheScreen();
    expect(txtNew).toBeOnTheScreen();
    expect(buttonSizeGuide).toBeOnTheScreen();
    expect(footerDoris).toBeOnTheScreen();
  });

  it('should navigation to webview doris', () => {
    const { getByTestId } = render(<ButtonDoris
      productEan={eanValid}
      enabledBtnFullDoris
    />);

    const btnDoris = getByTestId('com.usereserva:id/button_doris');

    fireEvent.press(btnDoris);

    expect(mockedNavigate).toHaveBeenCalledTimes(1);
  });

  it('should not navigation to webview doris', () => {
    const { getByTestId } = render(<ButtonDoris
      productEan=""
      enabledBtnFullDoris
    />);

    const btnDoris = getByTestId('com.usereserva:id/button_doris');

    fireEvent.press(btnDoris);

    expect(mockedNavigate).toHaveBeenCalledTimes(0);
    expect(mockedNavigate).not.toHaveBeenCalled();
  });
});
