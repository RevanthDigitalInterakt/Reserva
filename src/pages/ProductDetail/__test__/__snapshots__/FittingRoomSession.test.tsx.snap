// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FittingRoomSession should snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/fitting_room_session"
  testID="com.usereserva:id/fitting_room_session"
>
  <View
    backgroundColor="divider"
    height={1}
    my="xs"
    style={
      {
        "backgroundColor": "divider",
        "height": 1,
        "marginBottom": "xs",
        "marginTop": "xs",
      }
    }
    variant="fullWidth"
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityLabel="com.usereserva:id/component_button_doris"
      style={null}
      testID="com.usereserva:id/component_button_doris"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E4002B",
            "borderRadius": 4,
            "height": 19,
            "justifyContent": "center",
            "left": "-2.5%",
            "position": "absolute",
            "top": "-20%",
            "width": 47,
            "zIndex": 5,
          }
        }
      >
        <Text
          accessibilityLabel="com.usereserva:id/txt_new"
          style={
            {
              "color": "#ffffff",
              "fontFamily": "ReservaSans-Medium",
              "fontSize": 10,
            }
          }
          testID="com.usereserva:id/txt_new"
        >
          NOVO
        </Text>
      </View>
      <View
        accessibilityLabel="com.usereserva:id/button_doris"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="com.usereserva:id/button_doris"
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#000000",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "gap": 6,
              "height": 42,
              "justifyContent": "center",
              "width": 337.5,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="15"
            bbWidth="24"
            fill="none"
            focusable={false}
            height={15}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 15,
                  "width": 24,
                },
              ]
            }
            vbHeight={15}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M10.221 3.75c0-.778.435-2.333 2.173-2.333 1.739 0 2.174 1.166 2.174 2.333 0 1.167-2.17 1.46-2.174 2.917 0 0 8.497 2.457 9.78 2.916 1.284.46 1.087.584 1.087 2.334 0 1.166-.724 1.555-1.087 1.75h-3.26c-2.173 0-3.499.12-6.52 0-2.818.08-4.346 0-6.52 0h-3.26c-.362 0-1.086-.584-1.086-1.75 0-1.75 0-1.816 1.086-2.334 1.087-.517 7.607-2.333 7.607-2.333"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                  ]
                }
                stroke={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeWidth="1.16667"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              {
                "fontFamily": "Nunito-Regular",
                "fontSize": 25.78125,
                "textTransform": "uppercase",
              }
            }
          >
            vista em vocÃª
          </Text>
        </View>
      </View>
    </View>
    <View
      style={{}}
    >
      <View>
        <View
          accessibilityLabel="com.usereserva:id/size_guides_button_product_details"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="com.usereserva:id/size_guides_button_product_details"
        >
          <View
            style={
              {
                "alignItems": "center",
                "borderColor": "#000000",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "gap": 6,
                "height": 42,
                "justifyContent": "center",
                "width": 337.5,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="16"
              bbWidth="20"
              fill="none"
              focusable={false}
              height={16}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 16,
                    "width": 20,
                  },
                ]
              }
              vbHeight={16}
              vbWidth={20}
              width={20}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGMask
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  height="16"
                  maskContentUnits={1}
                  maskUnits={1}
                  name="a"
                  propList={
                    [
                      "fill",
                    ]
                  }
                  width="20"
                  x="-0.326416"
                  y="0.020752"
                >
                  <RNSVGPath
                    d="M-0.326416 0.020752H19.673584V16.020752H-0.326416z"
                    fill={
                      {
                        "payload": 4294967295,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M12.99 1.02c-3.608 0-6.434 1.426-6.434 3.485 0 1.295 1.118 2.413 2.844 3.064H1.867a1.268 1.268 0 00-.974.344.838.838 0 00-.217.625v5.391c.004.31.124.607.338.84.213.234.507.388.83.435l.026.004H14.5a5.528 5.528 0 001.969-.348 4.535 4.535 0 001.997-1.469.307.307 0 00.068-.086c.579-.782.89-1.715.89-2.671v-6.13c0-2.058-2.826-3.483-6.434-3.483zm5.794 9.614c0 .429-.073.855-.215 1.262v-.208a.312.312 0 00-.32-.303.312.312 0 00-.319.303v1.371c-.342.434-.779.793-1.28 1.051V11.69a.312.312 0 00-.32-.304.312.312 0 00-.32.303v2.685a4.772 4.772 0 01-1.04.206v-1.41a.312.312 0 00-.32-.304.312.312 0 00-.32.303v1.419l.001.014h-1.044l.002-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.32.303v2.159l.002.014h-1.044l.001-.014v-1.419a.312.312 0 00-.32-.303.312.312 0 00-.319.303v1.419l.002.014H9.927l.001-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.32.303v2.159l.002.014H8.246l.002-.014v-1.419a.312.312 0 00-.32-.303.312.312 0 00-.32.303v1.419l.002.014H6.566l.001-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.319.303v2.159l.002.014H4.886l.001-.014v-1.419a.312.312 0 00-.32-.303.312.312 0 00-.319.303v1.419l.002.014H3.205l.002-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.32.303v2.159l.002.014H1.93a.748.748 0 01-.434-.231.681.681 0 01-.181-.44V8.521l-.002-.034a.249.249 0 01.065-.18.706.706 0 01.489-.132H12.99c1.75 0 3.383-.388 4.598-1.092a.295.295 0 00.103-.415.33.33 0 00-.436-.103c-1.1.637-2.655 1.003-4.265 1.003-3.195 0-5.794-1.376-5.794-3.065 0-1.688 2.6-2.877 5.796-2.877s5.793 1.188 5.793 2.878v6.129z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M6.556 4.505v3.411h.64V4.504h-.64zM13.174 2.714c-2.337 0-4.167.843-4.167 1.917s1.83 1.917 4.167 1.917c2.336 0 4.166-.842 4.166-1.917 0-1.074-1.83-1.917-4.166-1.917zm0 3.39c-1.94 0-3.58-.674-3.58-1.473 0-.798 1.64-1.473 3.58-1.473 1.94 0 3.578.675 3.578 1.473 0 .799-1.638 1.473-3.578 1.473z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGMask>
                <RNSVGPath
                  d="M12.99 1.02c-3.608 0-6.434 1.426-6.434 3.485 0 1.295 1.118 2.413 2.844 3.064H1.867a1.268 1.268 0 00-.974.344.838.838 0 00-.217.625v5.391c.004.31.124.607.338.84.213.234.507.388.83.435l.026.004H14.5a5.528 5.528 0 001.969-.348 4.535 4.535 0 001.997-1.469.307.307 0 00.068-.086c.579-.782.89-1.715.89-2.671v-6.13c0-2.058-2.826-3.483-6.434-3.483zm5.794 9.614c0 .429-.073.855-.215 1.262v-.208a.312.312 0 00-.32-.303.312.312 0 00-.319.303v1.371c-.342.434-.779.793-1.28 1.051V11.69a.312.312 0 00-.32-.304.312.312 0 00-.32.303v2.685a4.772 4.772 0 01-1.04.206v-1.41a.312.312 0 00-.32-.304.312.312 0 00-.32.303v1.419l.001.014h-1.044l.002-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.32.303v2.159l.002.014h-1.044l.001-.014v-1.419a.312.312 0 00-.32-.303.312.312 0 00-.319.303v1.419l.002.014H9.927l.001-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.32.303v2.159l.002.014H8.246l.002-.014v-1.419a.312.312 0 00-.32-.303.312.312 0 00-.32.303v1.419l.002.014H6.566l.001-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.319.303v2.159l.002.014H4.886l.001-.014v-1.419a.312.312 0 00-.32-.303.312.312 0 00-.319.303v1.419l.002.014H3.205l.002-.014v-2.159a.312.312 0 00-.32-.303.312.312 0 00-.32.303v2.159l.002.014H1.93a.748.748 0 01-.434-.231.681.681 0 01-.181-.44V8.521l-.002-.034a.249.249 0 01.065-.18.706.706 0 01.489-.132H12.99c1.75 0 3.383-.388 4.598-1.092a.295.295 0 00.103-.415.33.33 0 00-.436-.103c-1.1.637-2.655 1.003-4.265 1.003-3.195 0-5.794-1.376-5.794-3.065 0-1.688 2.6-2.877 5.796-2.877s5.793 1.188 5.793 2.878v6.129z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M6.556 4.505v3.411h.64V4.504h-.64zM13.174 2.714c-2.337 0-4.167.843-4.167 1.917s1.83 1.917 4.167 1.917c2.336 0 4.166-.842 4.166-1.917 0-1.074-1.83-1.917-4.166-1.917zm0 3.39c-1.94 0-3.58-.674-3.58-1.473 0-.798 1.64-1.473 3.58-1.473 1.94 0 3.578.675 3.578 1.473 0 .799-1.638 1.473-3.578 1.473z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M9.4 7.57v.15l.053-.29-.053.14zm-7.533 0l-.012.15h.012v-.15zm-.974.343l-.105-.108-.007.008.112.1zm-.217.625h.15v-.012l-.15.012zm0 5.391h-.15v.002l.15-.002zm.338.84l.11-.1-.11.1zm.83.435l.024-.148h-.003l-.022.148zm.026.004l-.024.148.012.002h.012v-.15zm12.63 0v-.15.15zm1.928-.333l-.045-.143-.007.002.052.141zm.04-.015l-.053-.14-.007.003.06.137zm.378-.16l.064.136h.001l-.065-.136zm1.62-1.31l-.1-.112-.01.01-.009.011.12.092zm.068-.085l-.12-.09-.007.01-.006.01.133.07zm.89-2.671h.15-.15zm-.64 0h.15-.15zm-.215 1.262h-.15l.292.05-.142-.05zm-.639 1.163l.118.093.032-.04v-.053h-.15zm-1.28 1.051h-.15v.246l.218-.112-.069-.134zm-.64.263l.045.143.105-.033v-.11h-.15zm-1.04.206h-.15v.163l.162-.013-.013-.15zm-.639.022v.15h.155l-.005-.155-.15.005zm-1.044 0h-.15v.15h.15v-.15zm-.636 0v.15h.155l-.005-.155-.15.005zm-1.044 0h-.15v.15h.15v-.15zm-.636 0v.15h.155l-.005-.155-.15.005zm-1.044 0h-.15v.15h.15v-.15zm-.637 0v.15h.155l-.005-.155-.15.005zm-1.044 0h-.15v.15h.15v-.15zm.002-.014h-.15.15zm0-1.419h.15-.15zm-.638 1.433v.15h.155l-.005-.155-.15.005zm-1.044 0h-.15v.15h.15v-.15zm-.636 0v.15h.155l-.005-.155-.15.005zm-1.044 0h-.15v.15h.15v-.15zm.001-.014h-.15.15zm0-1.419h.15-.15zm-.637 1.433v.15h.155l-.006-.155-.15.005zm-1.045 0h-.15v.15h.15v-.15zm.002-.014h-.15.15zm0-2.159h.15-.15zm-.638 2.173v.15h.155l-.005-.155-.15.005zm-.639 0l-.024.148.012.002h.012v-.15zm-.434-.231l-.11.1.11-.1zm-.181-.44h-.15v.004l.15-.005zm0-5.408h.15v-.01l-.15.01zm-.002-.034l-.15.008v.002l.15-.01zm.065-.18l-.089-.122-.012.01-.01.01.11.101zm.489-.132l-.016.15h.016v-.15zm15.72-1.092l.076.13.002-.001-.077-.13zm-.332-.518l-.073-.131h-.002l.075.13zm-4.263-4.939v-.15.15zm5.793 2.878h.15-.15zM6.556 7.916h-.15v.15h.15v-.15zm.64 0v.15h.15v-.15h-.15zm9.556-3.285h-.15.15zM12.99.871c-1.821 0-3.455.36-4.64.987-1.182.627-1.944 1.54-1.944 2.647h.3c0-.951.652-1.781 1.785-2.382 1.13-.599 2.713-.952 4.5-.952v-.3zM6.406 4.505c0 1.39 1.195 2.545 2.941 3.205l.106-.28c-1.707-.645-2.747-1.726-2.747-2.925h-.3zM9.4 7.419H1.867v.3H9.4v-.3zm-7.522 0c-.404-.03-.803.11-1.09.386l.209.216c.223-.215.537-.327.858-.302l.023-.3zM.78 7.814a.988.988 0 00-.255.736l.3-.023a.688.688 0 01.178-.512l-.223-.201zm-.255.725v5.391h.3V8.538h-.3zm0 5.393c.005.347.139.68.377.94l.221-.203a1.119 1.119 0 01-.298-.74l-.3.003zm.377.94c.238.26.564.43.919.481l.043-.297a1.246 1.246 0 01-.74-.386l-.222.202zm.916.48l.027.005.05-.296-.028-.004-.05.296zm.051.007H14.5v-.3H1.87v.3zm12.628 0a5.677 5.677 0 001.981-.342l-.103-.282a5.378 5.378 0 01-1.876.324l-.002.3zm1.975-.34a.508.508 0 00.056-.021l-.121-.274a.234.234 0 01-.025.009l.09.286zm.049-.018c.13-.05.26-.104.388-.164l-.128-.271a5.134 5.134 0 01-.366.155l.106.28zm.389-.165a4.685 4.685 0 001.674-1.353l-.238-.183c-.41.533-.946.966-1.567 1.267l.131.27zm1.654-1.332a.455.455 0 00.101-.128l-.265-.14a.156.156 0 01-.034.043l.198.225zm.09-.109a4.644 4.644 0 00.919-2.76h-.3c0 .923-.3 1.825-.86 2.582l.24.178zm.919-2.76v-6.13h-.3v6.13h.3zm0-6.13c0-1.107-.762-2.019-1.944-2.646-1.184-.628-2.818-.987-4.64-.987v.3c1.787 0 3.37.353 4.5.952 1.132.6 1.784 1.43 1.784 2.382h.3zm-.94 6.13c0 .412-.07.821-.206 1.213l.283.098c.147-.423.223-.865.223-1.311h-.3zm.085 1.262v-.208h-.3v.208h.3zm0-.208a.462.462 0 00-.47-.453v.3c.102 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.469.454h.3c0-.078.069-.154.17-.154v-.3zm-.469.454v1.37h.3v-1.37h-.3zm.032 1.277a3.759 3.759 0 01-1.232 1.01l.138.268c.52-.269.974-.64 1.33-1.092l-.236-.186zM16.8 14.11V11.69h-.3v2.421h.3zm0-2.421a.462.462 0 00-.47-.454v.3c.102 0 .17.076.17.153h.3zm-.47-.454a.462.462 0 00-.469.454h.3c0-.078.068-.154.17-.154v-.3zm-.469.454v2.684h.3V11.69h-.3zm.105 2.541c-.327.104-.665.17-1.008.2l.025.299c.365-.031.725-.102 1.073-.213l-.09-.286zm-.846.35v-1.412h-.3v1.411h.3zm0-1.412a.462.462 0 00-.47-.453v.3c.102 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.47.453h.3c0-.077.07-.153.17-.153v-.3zm-.47.453v1.419h.3v-1.419h-.3zm0 1.419l.002.02.001.01c0 .002 0 0 0 0v-.012l.3-.01-.002-.017-.001-.007v.016h-.3zm.152-.136h-1.044v.3h1.044v-.3zm-.894.15v.014l-.001.003.001-.01a.18.18 0 00.002-.021h-.3a.127.127 0 010-.008.18.18 0 00-.002.022h.3zm.002-.014v-2.159h-.3v2.159h.3zm0-2.159a.462.462 0 00-.47-.453v.3c.101 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.47.453h.3c0-.077.069-.153.17-.153v-.3zm-.47.453v2.159h.3v-2.159h-.3zm0 2.159l.002.02v.01c.001.002.001 0 0 0v-.012l.3-.01-.002-.017v-.007.016h-.3zm.152-.136h-1.044v.3h1.044v-.3zm-.894.15l-.001.014v-.007c.002-.004.002-.012.002-.021h-.3l.001-.015v-.002.01a.18.18 0 00-.002.02h.3zm.001-.014v-1.419h-.3v1.419h.3zm0-1.419a.462.462 0 00-.47-.453v.3c.102 0 .17.075.17.153h.3zm-.47-.453a.462.462 0 00-.469.453h.3c0-.077.069-.153.17-.153v-.3zm-.469.453v1.419h.3v-1.419h-.3zm0 1.419l.001.02.002.01c0 .002 0 0 0 0l-.001-.012.3-.01a.2.2 0 00-.002-.017v-.004.013h-.3zm.152-.136H9.927v.3h1.044v-.3zm-.894.15l-.001.014v-.007l.002-.021h-.3v-.015l.001-.002a.229.229 0 00-.002.03h.3zm.001-.014v-2.159h-.3v2.159h.3zm0-2.159a.462.462 0 00-.47-.453v.3c.101 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.47.453h.3c0-.077.07-.153.17-.153v-.3zm-.47.453v2.159h.3v-2.159h-.3zm0 2.159l.002.02.001.01c0 .002 0 0 0 0a.11.11 0 010-.012l.3-.01a.18.18 0 00-.003-.024v.016h-.3zm.152-.136H8.246v.3H9.29v-.3zm-.894.15v.014l-.001.003.001-.01.002-.021h-.3v-.015-.002a.229.229 0 00-.002.03h.3zm.002-.014v-1.419h-.3v1.419h.3zm0-1.419a.462.462 0 00-.47-.453v.3c.101 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.47.453h.3c0-.077.07-.153.17-.153v-.3zm-.47.453v1.419h.3v-1.419h-.3zm0 1.419l.002.02.001.01c0 .002 0 0 0 0v-.012l.299-.01-.001-.017-.001-.007v.003a.12.12 0 010 .013h-.3zm.152-.136H6.566v.3H7.61v-.3zm-.894.15v.014l-.001.003.001-.01.001-.021h-.3l.001-.015v-.002.01a.178.178 0 00-.002.02h.3zm.001-.014v-2.159h-.3v2.159h.3zm0-2.159a.462.462 0 00-.47-.453v.3c.102 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.469.453h.3c0-.077.069-.153.17-.153v-.3zm-.469.453v2.159h.3v-2.159h-.3zm0 2.159l.001.02.002.01c0 .002 0 0 0 0l-.001-.012.3-.01-.002-.017v-.004.013h-.3zm.152-.136H4.886v.3H5.93v-.3zm-.894.15l-.001.014v-.007l.002-.021h-.3a.12.12 0 01.001-.015v-.002a.229.229 0 00-.002.03h.3zm.001-.014v-1.419h-.3v1.419h.3zm0-1.419a.462.462 0 00-.47-.453v.3c.102 0 .17.075.17.153h.3zm-.47-.453a.462.462 0 00-.469.453h.3c0-.077.068-.153.17-.153v-.3zm-.469.453v1.419h.3v-1.419h-.3zm0 1.419l.001.02.001.01c0 .002 0 0 0 0a.11.11 0 010-.012l.3-.01a.18.18 0 00-.003-.024v.003l.001.013h-.3zm.152-.136H3.205v.3H4.25v-.3zm-.895.15V14.608l.002-.021h-.3v-.015l.001-.002a.212.212 0 00-.003.03h.3zm.002-.014v-2.159h-.3v2.159h.3zm0-2.159a.462.462 0 00-.47-.453v.3c.102 0 .17.076.17.153h.3zm-.47-.453a.462.462 0 00-.47.453h.3c0-.077.07-.153.17-.153v-.3zm-.47.453v2.159h.3v-2.159h-.3zm0 2.159l.002.02.001.01c0 .002 0 0 0 0v-.012l.3-.01-.002-.017-.001-.007v.016h-.3zm.152-.136H1.93v.3h.64v-.3zm-.615.002a.598.598 0 01-.348-.185l-.22.203c.135.148.32.246.52.278l.048-.296zm-.348-.185a.532.532 0 01-.141-.343l-.3.009c.006.199.084.389.22.537l.221-.203zm-.141-.339V8.522h-.3v5.407h.3zm0-5.417l-.003-.034-.3.02.003.035.3-.02zm-.003-.032a.099.099 0 01.027-.072l-.222-.202a.398.398 0 00-.104.29l.3-.016zm.005-.052a.556.556 0 01.384-.103l.03-.298a.856.856 0 00-.592.16l.178.241zm.4-.102H12.99v-.3H1.867v.3zm11.123 0c1.77 0 3.43-.392 4.673-1.113l-.15-.259c-1.188.688-2.793 1.072-4.523 1.072v.3zm4.675-1.113a.445.445 0 00.152-.626l-.252.163a.145.145 0 01-.054.205l.154.258zm.152-.626a.48.48 0 00-.636-.152l.148.261a.18.18 0 01.236.054l.252-.163zm-.637-.151c-1.072.62-2.6.983-4.19.983v.3c1.63 0 3.212-.37 4.34-1.024l-.15-.26zm-4.19.983c-1.578 0-3.001-.34-4.025-.882-1.03-.545-1.62-1.271-1.62-2.033h-.3c0 .928.712 1.734 1.78 2.298 1.075.569 2.548.917 4.165.917v-.3zM7.346 4.504c0-.757.584-1.438 1.615-1.937 1.024-.496 2.449-.79 4.03-.79v-.3c-1.614 0-3.086.3-4.161.82-1.068.518-1.784 1.276-1.784 2.207h.3zm5.646-2.727c1.581 0 3.006.294 4.03.79 1.03.5 1.613 1.18 1.613 1.938h.3c0-.932-.715-1.69-1.783-2.208-1.074-.52-2.546-.82-4.16-.82v.3zm5.643 2.728v6.129h.3v-6.13h-.3zM6.706 7.916V4.505h-.3v3.411h.3zm-.15-3.261h.64v-.3h-.64v.3zm.49-.15v3.411h.3V4.504h-.3zm.15 3.261h-.64v.3h.64v-.3zm5.978-5.202c-1.185 0-2.249.214-3.023.57-.762.35-1.294.868-1.294 1.497h.3c0-.444.383-.886 1.12-1.225.725-.333 1.744-.542 2.897-.542v-.3zM8.857 4.631c0 .63.532 1.147 1.294 1.498.774.356 1.838.569 3.023.569v-.3c-1.153 0-2.172-.208-2.897-.542-.737-.339-1.12-.78-1.12-1.225h-.3zm4.317 2.067c1.184 0 2.248-.213 3.022-.57.762-.35 1.294-.867 1.294-1.497h-.3c0 .445-.383.886-1.12 1.225-.725.334-1.744.542-2.896.542v.3zM17.49 4.63c0-.63-.531-1.147-1.294-1.497-.774-.356-1.838-.57-3.022-.57v.3c1.152 0 2.171.208 2.896.542.737.339 1.12.78 1.12 1.225h.3zm-4.316 1.323c-.955 0-1.83-.166-2.46-.426-.316-.13-.563-.28-.73-.438-.165-.159-.24-.314-.24-.459h-.3c0 .255.132.483.334.676.201.192.485.36.822.499.673.277 1.588.448 2.574.448v-.3zm-3.43-1.323c0-.145.075-.3.24-.458.167-.159.414-.31.73-.44.63-.259 1.505-.425 2.46-.425v-.3c-.986 0-1.9.171-2.574.448-.337.139-.62.307-.822.5-.202.192-.333.42-.333.675h.3zm3.43-1.323c.954 0 1.829.166 2.46.426.315.13.562.28.728.439.166.158.24.313.24.458h.3c0-.254-.13-.483-.332-.675-.202-.193-.486-.361-.822-.5-.674-.277-1.589-.448-2.574-.448v.3zm3.428 1.323c0 .145-.073.3-.24.459-.165.158-.413.309-.729.438-.63.26-1.505.426-2.46.426v.3c.986 0 1.9-.17 2.575-.448.336-.139.62-.307.822-.5.201-.192.332-.42.332-.675h-.3z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  mask="a"
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <Text
              style={
                {
                  "fontFamily": "Nunito-Regular",
                  "fontSize": 25.78125,
                  "textTransform": "uppercase",
                }
              }
            >
              guia de medidas
            </Text>
          </View>
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
            "elevation": 5,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "zIndex": 5,
          }
        }
        transparent={true}
        visible={false}
      />
    </View>
  </View>
  <View
    accessibilityLabel="com.usereserva:id/footer_doris"
    style={
      {
        "marginTop": 16,
      }
    }
    testID="com.usereserva:id/footer_doris"
  >
    <Text
      style={
        {
          "fontFamily": "ReservaSans-Regular",
          "fontSize": 28.125,
          "textAlign": "center",
        }
      }
    >
      Vista a roupa que estÃ¡ vendo
       
      <Text
        style={
          {
            "fontFamily": "ReservaSans-Bold",
            "fontSize": 28.125,
            "fontWeight": "500",
          }
        }
      >
        em vocÃª
      </Text>
       
      ou em modelos com corpos similares ao seu
       
      <Text>
        ou se preferir, consulte o guia de medidas.
      </Text>
    </Text>
  </View>
  <View
    backgroundColor="divider"
    height={1}
    my="xs"
    style={
      {
        "backgroundColor": "divider",
        "height": 1,
        "marginBottom": "xs",
        "marginTop": "xs",
      }
    }
    variant="fullWidth"
  />
</View>
`;
